packages/react-instantsearch/src/connectors/connectMenu.js

```diff
@@ -28,7 +28,7 @@ function getValue(name, props, searchState) {
   return name === currentRefinement ? '' : name;
 }
 
-const sortBy = ['count:desc', 'name:asc'];
+const sortBy = ['isRefined', 'count:desc', 'name:asc'];
```

Comment:
Yes you're right. In any case we should definitely do something about this proposal https://github.com/algolia/instantsearch.js/issues/1734 (transformedProvidedProps), as it will helps to have both behavior easily. 
Also, if we sort by `isRefined` first, it will be a breaking change here.